### ISSUE TITLE
IPsrcaddr can't stop

### ISSUE THREAD

**Issue by underoll** [ClusterLabs/resource-agents#1128]
`ip route replace` in start and stop actions leaves the route without `proto kernel`
but the line https://github.com/ClusterLabs/resource-agents/blob/f35491fd18693d2816ad6f83c32e133b26193aa2/heartbeat/IPsrcaddr#L472 uses it while determining the route

looks like the commit https://github.com/ClusterLabs/resource-agents/commit/f35491fd18693d2816ad6f83c32e133b26193aa2 breaks the resource agent

i tried to use the agent in CentOS 7.4

Some diagnostic below
```
[root@node1 heartbeat]# ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: ens192: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
    link/ether 00:50:56:bf:9c:64 brd ff:ff:ff:ff:ff:ff
    inet 192.168.75.26/24 brd 192.168.75.255 scope global ens192
       valid_lft forever preferred_lft forever
    inet 192.168.75.42/24 brd 192.168.75.255 scope global secondary ens192:vip
       valid_lft forever preferred_lft forever
[root@node1 heartbeat]# ip ro
default via 192.168.75.1 dev ens192 src 192.168.75.42 
192.168.75.0/24 dev ens192 scope link src 192.168.75.42 
[root@node1 heartbeat]# pcs resource show cluster_ip
 Resource: cluster_ip (class=ocf provider=heartbeat type=IPaddr2)
  Attributes: iflabel=vip ip=192.168.75.42
  Operations: monitor interval=10s timeout=20s (cluster_ip-monitor-interval-10s)
              start interval=0s timeout=20s (cluster_ip-start-interval-0s)
              stop interval=0s timeout=20s (cluster_ip-stop-interval-0s)
[root@node1 heartbeat]# pcs resource show cluster_src
 Resource: cluster_src (class=ocf provider=heartbeat type=IPsrcaddr)
  Attributes: ipaddress=192.168.75.42
  Operations: monitor interval=10 timeout=20s (cluster_src-monitor-interval-10)
              start interval=0s timeout=20s (cluster_src-start-interval-0s)
              stop interval=0s timeout=20s (cluster_src-stop-interval-0s)
[root@node1 heartbeat]# pcs resource debug-stop cluster_src
Error performing operation: Operation not permitted
Operation stop for cluster_src (ocf:heartbeat:IPsrcaddr) returned 1
 >  stderr: Usage: ip route { list | flush } SELECTOR
 >  stderr:        ip route save SELECTOR
 >  stderr:        ip route restore
 >  stderr:        ip route showdump
 >  stderr:        ip route get ADDRESS [ from ADDRESS iif STRING ]
 >  stderr:                             [ oif STRING ]  [ tos TOS ]
 >  stderr:                             [ mark NUMBER ]
 >  stderr:        ip route { add | del | change | append | replace } ROUTE
 >  stderr: SELECTOR := [ root PREFIX ] [ match PREFIX ] [ exact PREFIX ]
 >  stderr:             [ table TABLE_ID ] [ proto RTPROTO ]
 >  stderr:             [ type TYPE ] [ scope SCOPE ]
 >  stderr: ROUTE := NODE_SPEC [ INFO_SPEC ]
 >  stderr: NODE_SPEC := [ TYPE ] PREFIX [ tos TOS ]
 >  stderr:              [ table TABLE_ID ] [ proto RTPROTO ]
 >  stderr:              [ scope SCOPE ] [ metric METRIC ]
 >  stderr: INFO_SPEC := NH OPTIONS FLAGS [ nexthop NH ]...
 >  stderr: NH := [ via ADDRESS ] [ dev STRING ] [ weight NUMBER ] NHFLAGS
 >  stderr: OPTIONS := FLAGS [ mtu NUMBER ] [ advmss NUMBER ]
 >  stderr:            [ rtt TIME ] [ rttvar TIME ] [reordering NUMBER ]
 >  stderr:            [ window NUMBER ] [ cwnd NUMBER ] [ initcwnd NUMBER ]
 >  stderr:            [ ssthresh NUMBER ] [ realms REALM ] [ src ADDRESS ]
 >  stderr:            [ rto_min TIME ] [ hoplimit NUMBER ] [ initrwnd NUMBER ]
 >  stderr:            [ features FEATURES ] [ quickack BOOL ] [ congctl NAME ]
 >  stderr:            [ expires TIME ]
 >  stderr: TYPE := { unicast | local | broadcast | multicast | throw |
 >  stderr:           unreachable | prohibit | blackhole | nat }
 >  stderr: TABLE_ID := [ local | main | default | all | NUMBER ]
 >  stderr: SCOPE := [ host | link | global | NUMBER ]
 >  stderr: NHFLAGS := [ onlink | pervasive ]
 >  stderr: RTPROTO := [ kernel | boot | static | NUMBER ]
 >  stderr: TIME := NUMBER[s|ms]
 >  stderr: BOOL := [1|0]
 >  stderr: FEATURES := ecn
 >  stderr: ocf-exit-reason:command 'ip route replace  dev ens192' failed
```

**Comment by underoll** [comment]
Should be fixed by 6052e8fd37d23f46db217f915b445c7e67dccb34

**Comment by oalbrigt** [comment]
Thanks for the update.
